bugsnag-android-core/src/main/java/com/bugsnag/android/ErrorReader.java

```diff
@@ -238,40 +239,41 @@ private static BugsnagException readException(JsonReader reader) throws IOExcept
     }
 
 
-    private static StackTraceElement[] readStackFrames(JsonReader reader) throws IOException {
-        ArrayList<StackTraceElement> frames = new ArrayList<>();
+    private static List<Map<String, Object>> readStackFrames(JsonReader reader) throws IOException {
+        List<Map<String, Object>> frames = new ArrayList<>();
         reader.beginArray();
         while (reader.hasNext()) {
             frames.add(readStackFrame(reader));
         }
         reader.endArray();
-        return frames.toArray(new StackTraceElement[0]);
+        return frames;
     }
 
-    private static StackTraceElement readStackFrame(JsonReader reader) throws IOException {
-        String method = null;
-        String file = null;
-        int lineNumber = 0;
+    private static Map<String, Object> readStackFrame(JsonReader reader) throws IOException {
+        Map<String, Object> map = new HashMap<>();
         reader.beginObject();
+
         while (reader.hasNext()) {
-            switch (reader.nextName()) {
-                case "method":
-                    method = reader.nextString();
-                    break;
-                case "file":
-                    file = reader.nextString();
+            String key = reader.nextName();
+            Object val = null;
+
+            switch (reader.peek()) {
+                case STRING:
+                    val = reader.nextString();
```

Comment:
Its a bit more concise to put the value in the map immediately rather than checking afterwards that it is non-null. If the token type is `STRING` or `NUMBER` then we know the value isn't null.
```suggestion
                    map.put(key, reader.nextString());
```